tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -3812,33 +3812,45 @@ void TestConvertGather(OpConverterTest* test) {
 
   // Ok.
   for (int i = 0; i < kGatherOKCases; i++) {
-    test->Reset();
-    test->AddTestTensor("params", ok_params[i].params_dims, 1,
-                        TfDataTypeToTrt(dtype));
-    test->AddTestTensor("indices", ok_params[i].indices_dims, 1,
-                        nvinfer1::DataType::kINT32);
-    test->AddTestWeights<int32>("axis", {1}, {ok_params[i].axis});
-    test->RunValidationAndConversion(node_def);
-    TRT_TensorOrWeights output;
-    TF_EXPECT_OK(test->GetTensorOrWeights("my_gather", &output));
-    ASSERT_TRUE(output.is_tensor());
-    ExpectTrtDimsEqualsArray(ok_params[i].expected_output_dims,
-                             output.tensor()->getDimensions());
+    for (const bool params_is_tensor : {true, false}) {
+      test->Reset();
+      if (params_is_tensor) {
+        test->AddTestTensor("params", ok_params[i].params_dims, 1,
+                            TfDataTypeToTrt(dtype));
+      } else {
+        test->AddTestWeights<CType>("params", ok_params[i].params_dims,
```

Comment:
As mentioned above, I think the reason why the test doesn't fail is that we didn't add the batch_dim of value 1 as the first dimension of the weights.

Reply:
I need to add an additional test for that.
Do you think that I should add some additional boolean variable to the TestParams (for params weight) or I should do that in the same way as now?

Reply:
Doing the current way is fine. I think we can define a batch_dim variable and used it in both `AddTestTensor` and `AddTestWeights`. Thanks.